src/shared_utilities/drivers.jl

```diff
@@ -185,11 +187,170 @@ PrescribedPrecipitation{FT}(liquid_precip) where {FT} =
     PrescribedPrecipitation{FT, typeof(liquid_precip)}(liquid_precip)
 
 """
-    CoupledRadiativeFluxes{FT} <: AbstractRadiativeDrivers{FT}
+    CoupledRadiativeFluxes{
+        FT,
+        F <: Union{Function, Nothing},
+        T,
+        LT,
+        IP,
+    } <: AbstractRadiativeDrivers{FT}
+
+To be used when coupling to an atmosphere model. If `start_date`, `latitude`, `longitude`,
+or `insol_params` are not `nothing`, then θs must be a function.
+
+During the driver update, cosθs is unchanged if `θs` is `nothing`.
+Otherwise, `θs` recieves the following arguments:
+(field_to_update, time_from_start, `start_date`, `latitude`, `longitude`, `insol_params`),
+and is expected to update `field_to_update` with the cosine zenith angle at the given time.
+$(DocStringExtensions.FIELDS)
+"""
+struct CoupledRadiativeFluxes{FT, F <: Union{Function, Nothing}, T, LT, IP} <:
+       AbstractRadiativeDrivers{FT}
+    """Function that fills a climacore field with the zentih angle given the following arguments:
+    (field_to_update, time_from_start, start_date, `latitude`, `longitude`, `insol_params`)"""
+    θs::F
+    "Start date - the datetime corresponding to t=0 for the simulation"
+    start_date::T
+    "latitude or collection of latitudes to compute the zenith angle for"
+    latitude::LT
+    "longitude or collection of longitudes to compute the zenith angle for"
+    longitude::LT
+    "Insolation parameters used to compute the zenith angle"
+    insol_params::IP
+    function CoupledRadiativeFluxes{FT, F, T, LT, IP}(
+        θs::F,
+        start_date::T,
+        latitude::LT,
+        longitude::LT,
+        insol_params::IP,
+    ) where {FT, F, T, LT, IP}
+        (
+            all(isnothing, (start_date, latitude, longitude, insol_params)) ||
+            !isnothing(θs)
+        ) || error(
```

Comment:
And I think maybe here the `||` should be `&&`, so that this errors if the previous condition is met. I'm not sure though